baybe/searchspace/discrete.py

```diff
@@ -670,6 +671,28 @@ def transform(
         return comp_rep
 
 
+def _apply_polars_constraint_filter(
+    ldf: pl.LazyFrame, constraints: Collection[DiscreteConstraint]
+) -> pl.LazyFrame:
+    """Remove discrete search space entries inplace based on constraints.
+
+    Args:
+        ldf: The data in experimental representation to be modified inplace.
+        constraints: List of discrete constraints.
+
+    Returns:
+        A Polars LazyFrame with undesired rows removed.
```

Comment:
going with `polars.objectname` everywhere, is that fine?

Reply:
Totally fine. Can you do it in a way that the doc generates the autolink to the polars page? I think that should automatically happen when you put it in backticks, I hope

Reply:
can you link me to an instance where we have this working? I included the name changes and ticks, but cannot get it to work with links to the other package

tried

- ``polars.LazyFrame.filter``
- :meth:`polars.LazyFrame.filter`
 - :meth:``polars.LazyFrame.filter``

Reply:
I'm quite sure that we have something like that somewhere but also can't remember 🤔 
What's the difference between the last two options you listed? Have you tried using the imported `pl` namespace?

Reply:
the difference is rendered awar by github (double vs single backticks)

but trying the namespace `pl` also doesnt work

I dont recall having this anywhere in our doc, except for python builtins

Reply:
Links to external packages should be generated automatically by sphinx. It might be necessary to add a link to polars to the `intersphinx mapping` in the `conf.py` file:
<img width="625" alt="image" src="https://github.com/emdgroup/baybe/assets/84016733/a30a2daa-30ba-44aa-afd4-365e59db34e1">

Reply:
thanks, will give it another try

Reply:
so I added one link, in other places theres no real opportunity and it seems the linking doesnt work with :class:`polars.Expr` and other classes, seems polars doc page handles top level class pages different than eg pandas and sphinx is not able to find it

Reply:
Well than people need to use google 😄

Reply:
Alright, no worries. Then let's resolve and move on 👍🏼